R-packages/evalcast/vignettes/intro_vignette.Rmd

```diff
@@ -0,0 +1,132 @@
+
+
+## Testing forecasters
+We will use the included `baseline_forecaster` to demonstrate how to backtest a forecaster and compare its predictions to contemporaneous forecasts submitted to CovidHub. While `baseline_forecaster` is very simple, it can be used as a template for creating more sophisticated forecasters.
+
+The format of a forecaster is aligned with the [CovidHub submission instructions](https://github.com/reichlab/covid19-forecast-hub/blob/master/data-processed/README.md) and with the [covidcast](https://cmu-delphi.github.io/covidcast/covidcastR/) R package.
+
+### Specify the signals that will be used by forecaster
+
+We start by specifying which signals the forecaster will be using.  See [here](https://cmu-delphi.github.io/delphi-epidata/api/covidcast_signals.html) for a full list of signals available through the `covidcast` API.  Each signal is specified by two strings: `data_source` and `signal`.  Optionally, we can specify how far back in time we will want data.
+
+```{r}
+library(tidyverse)
+signals <- tibble(data_source = "jhu-csse", 
+                  signal = c("deaths_incidence_num", "confirmed_incidence_num"), 
+                  start_day = "2020-06-15")
+signals
+```
+
+### Run the forecaster on some dates
+
+Within the `evalcast` framework, forecasters do not directly retrieve historic data. Instead, `get_predictions` provides *only the data that would have been available at the time* to the specified forecaster in order to generate retrospective predictions.
+
+```{r}
+library(evalcast)
+library(covidcast)
+library(lubridate)
+
+forecast_dates <- as_date(get_forecast_dates("CMU-TimeSeries"))
+forecast_dates_july <- forecast_dates[forecast_dates >= "2020-07-01" & forecast_dates <= "2020-07-31"]
+
+# Retrieve past predictions from CovidHub
+predictions_cards_yyg <- get_covidhub_predictions("YYG-ParamSearch", forecast_dates_july)
+predictions_cards_cmu <- get_covidhub_predictions("CMU-TimeSeries", forecast_dates_july)
+
+# Make new predictions using baseline_forecaster
+predictions_cards <- get_predictions(baseline_forecaster,
+                                     name_of_forecaster = "baseline",
+                                     signals = signals,
+                                     forecast_dates = forecast_dates_july,
+                                     incidence_period = "epiweek",
+                                     ahead = 3,
+                                     geo_type = "state")
+```
+
+`baseline_forecaster` created a list of `r length(predictions_cards)` predictions cards, (one for each `(forecast date, ahead)` pair).  Let's look at a predictions card:
+
+```{r}
+predictions_cards[[1]]
+```
```

Comment:
This would be a good place to demonstrate the function `plot_trajectory`.  Currently this is giving me an error when I try it on `predictions_cards`.  However, it works on `predictions_cards_cmu`.